### PR TITLE
[enhance][doc?] Add a note for 'distro' attribute in .travis.yml

### PR DESCRIPTION
**Pull request by 130s**
Inspired by #167

### CODE REVIEW

**Comment by 130s** [comment]
This `.travis.yml` file is referenced from the [readme](https://github.com/ros-industrial/industrial_ci#concrete-examples-of-config-files) so hope this change without adding doc in any other location is enough.

**Comment by ipa-mdl** [comment]
According to the [travis docs](https://docs.travis-ci.com/user/docker/) the failsafe way is to specify.
```
sudo: required

services:
  - docker
````
This enforces to run the build on trusty
> Travis CI automatically routes builds to run on Trusty sudo: required when services: docker is configured.

**Comment by 130s** [comment]
Oh, okay, I didn't know that. But then, either `dist` or `services` must be specified if I understand correctly.
Since none of the [example config files listed in the readme](https://github.com/ros-industrial/industrial_ci/blob/a3336ff0ec0daf361ce881dd1951bdc113b0ac91/README.rst#id6) use this `services` attribute with `docker` so far, wouldn't it still reasonable to recommend `dist` (caveat is that when Travis updates their spec changes may have to be applied on all repositories)?

**Comment by ipa-mdl** [comment]
The code does not depend on a specific distro, but on docker.
I would suggest to include `dist` and `services`, the latter emphasizes that docker is required.

**Comment by 130s** [comment]
> I would suggest to include `dist` and `services`, the latter emphasizes that docker is required.

Ok, emphasizing that `docker` is required sounds much more beneficial.
Commit updated accordingly.

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,9 @@
 #  Greatly inspired by JSK travis: https://github.com/jsk-ros-pkg/jsk_travis
 
 sudo: required
-dist: trusty
+services:
+  - docker
+dist: trusty  # Optional, as long as "services" is set with "docker". This will have to be updated once Travis CI supports newer Ubuntu distros.
```
Comment (ipa-mdl) [inline .travis.yml]: > This will have to be updated once Travis CI supports newer Ubuntu distros.

There is no need to.

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,9 @@
 #  Greatly inspired by JSK travis: https://github.com/jsk-ros-pkg/jsk_travis
 
 sudo: required
```
Comment (ipa-mdl) [inline .travis.yml]: this is optional as well.

**Comment by 130s** [comment]
Actually I removed some optional attributes in https://github.com/ros-industrial/industrial_ci/pull/168/commits/28b05e78f9f6d2e65bf2a430564e492ef9a6c983. Repos that still use them should still function and there won't be any change required.

**Comment by 130s** [comment]
@ipa-mdl sorry I added another commit after getting your approval and take back my previous commit. As the commit message reads:

Since `.travis.yml` in this repo serves as reference for new users in addition to its normal purpose, it's worthwhile to leave them even commented out.

**Comment by ipa-mdl** [comment]
>  sorry I added another commit after getting your approval and take back my previous commit. 

I am still oky with it, so merge it if you are ready.

> Since .travis.yml in this repo serves as reference for new users in addition to its normal purpose

This needs to be changed, our `.travis.yml`is way too complicated for the regular users.
I'd like to introduce a new sample config in `doc/` with a lot of explanations.

**Comment by 130s** [comment]
Thanks for thoughtful comments as always @ipa-mdl.

> This needs to be changed, our .travis.ymlis way too complicated for the regular users.

So true. Current `.travis.yml` may not even function as a reference at all.

**Pull request merged**